source/server/hot_restarting_parent.cc

```diff
@@ -128,26 +128,26 @@ HotRestartingParent::Internal::getListenSocketsForChild(const HotRestartMessage:
 // magnitude of memory usage that they are meant to avoid, since this map holds full-string
 // names. The problem can be solved by splitting the export up over many chunks.
 void HotRestartingParent::Internal::exportStatsToChild(HotRestartMessage::Reply::Stats* stats) {
-  for (const auto& gauge : server_->stats().gauges()) {
-    if (gauge->used()) {
-      const std::string name = gauge->name();
-      (*stats->mutable_gauges())[name] = gauge->value();
-      recordDynamics(stats, name, gauge->statName());
+  server_->stats().forEachSinkedGauge(nullptr, [this, stats](Stats::Gauge& gauge) mutable {
```

Comment:
This occurs on every stats flush.

I think if there is a CPU problem with flushing these stats normally, there'll be the same problem (maybe even worse with some IPC messages needing to be passed) in the hot-restart case. 

I think you should clearly document that all metrics which are filtered for sinks will also not be sync'd during hot-restart, and note that this also applies to stats from the admin console. 

The alternative is to have a system that's working pretty well, and have it vulnerable to completely falling over when you try a hot-restart.

@mattklein123 wdyt?

Reply:
> This occurs on every stats flush.

Pedantically, it occurs on every flush for the duration of the hot restart drain time.

Sorry for missing this ping a few days ago. I tend to agree with @ggreenway that it would be better to avoid unsurprising behavior. Can we clearly document that we won't send non-sink stats over to the child and that means that admin will be wrong during hot restart? Maybe this documentation is minimal right now since we don't support API configuration of this feature and people using the API directly I guess would know what they are doing, though it might still be surprising. If we support API configuration of this feature I think the documentation bar would get higher with warnings in various places.